src/Microsoft.DotNet.Build.CloudTestTasks/AzureHelper.cs

```diff
@@ -0,0 +1,234 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Globalization;
+using System.Linq;
+using System.Net;
+using System.Security.Cryptography;
+using System.Text;
+using System.Text.RegularExpressions;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.DotNet.Build.CloudTestTasks
+{
+    public class AzureHelper
+    {
+        public const string STORAGE_API_VERSION = "2015-04-05";
+
+        public static string createContainerSASToken(string AccountName, string ContainerName, string key,
+            string accessType, int tokenValue)
+        {
+            string signedpermissions = "r";
+            if (accessType == "write")
+            {
+                signedpermissions += "wdl";
+            }
+            string signedstart = DateTime.UtcNow.ToString("O");
+            string signedexpiry = DateTime.UtcNow.AddDays(tokenValue).ToString("O");
+            string canonicalizedresource = "/blob/" + AccountName + "/" + ContainerName;
+            string signedidentifier = "";
+            string signedversion = "2015-04-05";
+            string rscc = "";
+            string rscd = "";
+            string rsce = "";
+            string rscl = "";
+            string rsct = "";
+
+
+            string StringToSign = signedpermissions + "\n" +
+               signedstart + "\n" +
+               signedexpiry + "\n" +
+               canonicalizedresource + "\n" +
+               signedidentifier + "\n" +
+               "" + "\n" +
+               "" + "\n" +
+               signedversion + "\n" +
+               rscc + "\n" +
+               rscd + "\n" +
+               rsce + "\n" +
+               rscl + "\n" +
+               rsct;
+
+            byte[] SignatureBytes = System.Text.Encoding.UTF8.GetBytes(StringToSign);
+            System.Security.Cryptography.HMACSHA256 SHA256 = new System.Security.Cryptography.HMACSHA256(Convert.FromBase64String(key));
```

Comment:
Also, this object is an "hmacSha256", not a "sha256" (there's a huge difference between the two)